test/e2e/tests/request-queuing/multi-dapp-sendTx-revokePermission.spec.js

```diff
@@ -95,26 +86,37 @@ describe('Request Queuing for Multiple Dapps and Txs on different networks revok
 
         // Dapp 1 send tx
         await driver.switchToWindowWithUrl(DAPP_URL);
-        await driver.delay(largeDelayMs);
+        await driver.findElement({
+          css: '[id="chainId"]',
+          text: '0x1',
+        });
         await driver.clickElement('#sendButton');
 
         await driver.waitUntilXWindowHandles(4);
         await driver.delay(3000);
 
         // Dapp 2 send tx
         await driver.switchToWindowWithUrl(DAPP_ONE_URL);
-        await driver.delay(largeDelayMs);
+        await driver.findElement({
+          css: '[id="chainId"]',
+          text: '0x53a',
+        });
         await driver.clickElement('#sendButton');
 
         // Dapp 1 revokePermissions
         await driver.switchToWindowWithUrl(DAPP_URL);
-        await driver.clickElement('#revokeAccountsPermission');
+        await driver.findElement({
+          css: '[id="chainId"]',
+          text: '0x1',
+        });
 
         // Confirmation will close then reopen
-        await driver.waitUntilXWindowHandles(3);
+        await driver.clickElement('#revokeAccountsPermission');
+        // TODO: find a better way to handle different dialog ids
+        await driver.delay(3000);
```

Comment:
the test failed at this point, as a dialog is closed and re-opened, giving space to race conditions where the context is invalidated